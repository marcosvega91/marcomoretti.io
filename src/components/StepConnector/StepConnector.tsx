import styled, { css } from 'styled-components';
import Orientation from '../Stepper/Orientation';

export interface StepperProps {
  orientation?: Orientation;
}

const horizontal = css`
  border-top-width: 1px;
  border-top-style: solid;
`;

const vertical = css`
  border-left-width: 1px;
  border-left-style: solid;
  min-height: 24px;
`;

const StepConnector = styled.div<StepperProps>`
  ${({ orientation }) => (orientation === 'vertical' ? vertical : horizontal)};
  border-color: aqua;
  flex: 1 1 auto;
  // To center with the circle of the step
  margin-top: 12px;
`;

StepConnector.defaultProps = {
  orientation: 'vertical',
};

export default StepConnector;
