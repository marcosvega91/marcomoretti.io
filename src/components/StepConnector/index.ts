import * as stepConnector from './StepConnector';
import StepConnector from './StepConnector';

export type StepperProps = stepConnector.StepperProps;
export default StepConnector;
