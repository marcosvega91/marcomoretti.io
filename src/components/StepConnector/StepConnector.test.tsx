import React from 'react';
import { render } from '@testing-library/react';
import StepConnector from './index';

describe('<StepConnector />', () => {
  it('should be vertical by default', () => {
    const { container } = render(<StepConnector />);
    const stepConnector = container.firstChild;
    expect(stepConnector).toMatchSnapshot();
  });

  it('should be horizontal when orientation="horizontal"', () => {
    const { container } = render(<StepConnector orientation="horizontal" />);
    const stepConnector = container.firstChild;
    expect(stepConnector).toMatchSnapshot();
  });

  it('should be horizontal when orientation="vertical"', () => {
    const { container } = render(<StepConnector orientation="vertical" />);
    const stepConnector = container.firstChild;
    expect(stepConnector).toMatchSnapshot();
  });
});
