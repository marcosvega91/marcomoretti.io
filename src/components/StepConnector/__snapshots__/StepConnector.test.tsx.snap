// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepConnector /> should be horizontal when orientation="horizontal" 1`] = `
.c0 {
  border-top-width: 1px;
  border-top-style: solid;
  border-color: black;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin-top: 12px;
}

<div
  class="c0"
  orientation="horizontal"
/>
`;

exports[`<StepConnector /> should be horizontal when orientation="vertical" 1`] = `
.c0 {
  border-left-width: 1px;
  border-left-style: solid;
  min-height: 24px;
  border-color: black;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin-top: 12px;
}

<div
  class="c0"
  orientation="vertical"
/>
`;

exports[`<StepConnector /> should be vertical by default 1`] = `
.c0 {
  border-left-width: 1px;
  border-left-style: solid;
  min-height: 24px;
  border-color: black;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin-top: 12px;
}

<div
  class="c0"
  orientation="vertical"
/>
`;
